### PR TITLE
BNCLog functions for Swift

### PR DESCRIPTION
**Pull request by jdee**
This is a Swift wrapper for the BNCLog Objective-C macros. It's not clear whether this should be integrated into the SDK, but it's useful and can at least serve as an example.

The BNCLogMessageInternal function is not available in Swift because it's variadic, so a BNCSwiftLog Objective-C class was provided with class methods to be called from Swift. For now, this requires a bridging header in the app. **Updated**: Another commit was added that just adds BNCLogMessageInternalSwift to BNCLog.[hm] in the SDK with a different signature. That allows removal of the bridging header and the Obj-C shim in the app. Of course, this commit can be reverted if it's not desirable to add such a function to the SDK. But this seems very simple by comparison with the previous approach.

Also, it was necessary to call `BNCLogSetOutputFunction(nil)` at initialization in order to avoid duplicate output.

To test, run the WebviewExample on a simulator and select any row from the table view. This will produce output like:

```
2017-04-07 10:25:00.909 WebViewExample[3386:69681] [branch.io] ArticleViewController.swift(92)     Log: Created Branch Universal Object
2017-04-07 10:25:01.425 WebViewExample[3386:69681] [branch.io] ArticleViewController.swift(67)     Log: Logged BNCRegisterViewEvent on BUO
```

### CODE REVIEW

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/0d965583a713312fcf00593f5c5d134d/0d965583a713312fcf00593f5c5d21a5)
---
*Duration*: 3 minutes and 39 seconds
*Result*: **1 error, failing state: build-errors**

File: Examples/WebViewExample/WebViewExample/Utilities/BNCSwiftLog.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//  Created by Jimmy Dee on 4/7/17.
+//  Copyright © 2017 Branch Metrics. All rights reserved.
+//
+
+@import Foundation;
+
+/**
+ * Swift-friendly Obj-C wrapper for BNCLogMessageInternal
+ */
+@interface BNCSwiftLog : NSObject
```
Comment (jdee) [inline Examples/WebViewExample/WebViewExample/Utilities/BNCSwiftLog.h]: This entire wrapper could be a single method that also takes a BNCLogLevel enum argument, like BNCLogMessageInternal itself. This was adapted from something else where this approach made more sense.

**Comment by BranchMacMini** [comment]
Result of [Integration 2](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/0d965583a713312fcf00593f5c5d134d/0d965583a713312fcf00593f5c77b7f3)
---
*Duration*: 2 minutes and 9 seconds
*Result*: **1 error, failing state: build-errors**

**Comment by BranchMacMini** [comment]
Result of [Integration 4](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/0d965583a713312fcf00593f5c5d134d/3798ddb0b089fd3023ef978e320086fc)
---
*Duration*: 2 minutes and 31 seconds
*Result*: **1 error, failing state: build-errors**

**Comment by BranchMacMini** [comment]
Result of [Integration 5](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/0d965583a713312fcf00593f5c5d134d/f0a4a200a17291163ccbc690d8378125)
---
*Duration*: 1 minute and 52 seconds
*Result*: **1 error, failing state: build-errors**

**Comment by BranchMacMini** [comment]
Result of [Integration 6](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/0d965583a713312fcf00593f5c5d134d/f0a4a200a17291163ccbc690d8db2a04)
---
*Duration*: 2 minutes and 26 seconds
*Result*: **Perfect build!** All 179 tests passed. :+1:
*Test Coverage*: 65%

**Comment by jdee** [comment]
One minor thing I noticed since I also wrapped the React Obj-C logger in the RN repo: The RCTLog macros/functions autocomplete better than the BNCLog ones. I think the reason is that there are various functions and other things that begin with BNCLog. By contrast, e.g., the RCTLog macros call `_RCTLogNativeInternal`, so that when you type RCTLog<ESC>, you only see the developer-facing symbols:

screen shot 2017-04-14 at 2 04 43 pm

screen shot 2017-04-14 at 2 03 32 pm

I assume it's noisier in Objective-C. I rely heavily on autocompletion in Xcode, so this was noticeable, but hardly a huge problem, especially for an internal utility. But renaming some things might make it a bit easier to use.

**Comment by BranchMacMini** [comment]
Result of [Integration 7](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/0d965583a713312fcf00593f5c5d134d/30100875d20faf229b7e6adcc8001b5f)
---
*Duration*: 6 minutes and 37 seconds
*Result*: **Perfect build!** All 179 tests passed. :+1:
*Test Coverage*: 65%

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/9d628bd21bc7d253a85fa498263d4200/9d628bd21bc7d253a85fa498263d64ab)
---
*Duration*: 1 minute and 20 seconds
*Result*: **-1 error, failing state: build-failed**

**Comment by BranchMacMini** [comment]
Result of [Integration 2](https://stlt.herokuapp.com/v1/xcs_deeplink/BranchMacMini.local/9d628bd21bc7d253a85fa498263d4200/9d628bd21bc7d253a85fa498263f92a2)
---
*Duration*: 19 seconds
*Result*: **-1 error, failing state: build-failed**

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/4671cb6175c07343ead50b38df1d0301/4671cb6175c07343ead50b38df1d0e5b)
---
*Duration*: 4 minutes and 47 seconds
*Result*: **1 error, failing state: build-errors**

**Comment by BranchMacMini** [comment]
Result of [Integration 2](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/4671cb6175c07343ead50b38df1d0301/4671cb6175c07343ead50b38df3d2505)
---
*Duration*: 3 minutes and 2 seconds
*Result*: **Perfect build!** All 179 tests passed. :+1:
*Test Coverage*: 65%

**Pull request merged**